src/common/meta/src/ddl/alter_table.rs

```diff
@@ -115,10 +117,16 @@ impl AlterTableProcedure {
                 let requester = requester.clone();
 
                 alter_region_tasks.push(async move {
-                    requester
-                        .handle(request)
-                        .await
-                        .map_err(add_peer_context_if_needed(datanode))
+                    if let Err(err) = requester.handle(request).await {
+                        if err.status_code() != StatusCode::RequestOutdated {
```

Comment:
I think we need a util method to handle these special errors. Perhaps not only "alter" will encounter this error, but other DDL operations may also need it (Such as logical alter .

Reply:
The metric table will check the `add columns` requests and only add the missing columns.